src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -2634,139 +2665,113 @@ bool FlexPA::genPatterns_commit(
+      via = std::make_unique<frVia>(access_point->getViaDef());
+      auto rvia = via.get();
+      temp_vias.push_back(std::move(via));
+
+      dbTransform xform = inst->getUpdatedXform(true);
+      Point pt(access_point->getPoint());
+      xform.apply(pt);
+      rvia->setOrigin(pt);
+      if (inst_term->hasNet()) {
+        objs.emplace_back(rvia, inst_term->getNet());
+      } else {
+        objs.emplace_back(rvia, inst_term);
+      }
+    }
   }
 
-  // add to pattern set if unique
-  if (inst_access_patterns.find(access_pattern) == inst_access_patterns.end()) {
-    inst_access_patterns.insert(access_pattern);
-    // create new access pattern and push to uniqueInstances
-    auto pin_access_pattern = std::make_unique<FlexPinAccessPattern>();
-    std::map<frMPin*, frAccessPoint*> pin_to_access_point;
-    // check DRC for the whole pattern
-    std::vector<std::pair<frConnFig*, frBlockObject*>> objs;
-    std::vector<std::unique_ptr<frVia>> temp_vias;
-    frInst* target_obj = nullptr;
-    for (int pin_idx = 0; pin_idx < (int) pins.size(); pin_idx++) {
-      auto acc_point_idx = access_pattern[pin_idx];
-      auto& [pin, inst_term] = pins[pin_idx];
-      auto inst = inst_term->getInst();
-      target_obj = inst;
-      const int pin_access_idx = unique_insts_.getPAIndex(inst);
-      const auto pa = pin->getPinAccess(pin_access_idx);
-      const auto access_point = pa->getAccessPoint(acc_point_idx);
-      pin_to_access_point[pin] = access_point;
-
-      // add objs
-      std::unique_ptr<frVia> via;
-      if (access_point->hasAccess(frDirEnum::U)) {
-        via = std::make_unique<frVia>(access_point->getViaDef());
-        auto rvia = via.get();
-        temp_vias.push_back(std::move(via));
+  frAccessPoint* leftAP = nullptr;
+  frAccessPoint* rightAP = nullptr;
+  frCoord leftPt = std::numeric_limits<frCoord>::max();
+  frCoord rightPt = std::numeric_limits<frCoord>::min();
 
-        dbTransform xform = inst->getUpdatedXform(true);
-        Point pt(access_point->getPoint());
-        xform.apply(pt);
-        rvia->setOrigin(pt);
-        if (inst_term->hasNet()) {
-          objs.emplace_back(rvia, inst_term->getNet());
-        } else {
-          objs.emplace_back(rvia, inst_term);
+  const auto& [pin, inst_term] = pins[0];
+  const auto inst = inst_term->getInst();
+  for (auto& inst_term : inst->getInstTerms()) {
+    if (isSkipInstTerm(inst_term.get())) {
+      continue;
+    }
+    uint64_t n_no_ap_pins = 0;
+    for (auto& pin : inst_term->getTerm()->getPins()) {
+      if (pin_to_access_point.find(pin.get()) == pin_to_access_point.end()) {
+        n_no_ap_pins++;
+        pin_access_pattern->addAccessPoint(nullptr);
+      } else {
+        const auto& ap = pin_to_access_point[pin.get()];
+        const Point tmpPt = ap->getPoint();
+        if (tmpPt.x() < leftPt) {
+          leftAP = ap;
+          leftPt = tmpPt.x();
+        }
+        if (tmpPt.x() > rightPt) {
```

Comment:
Should this be `else if`?

Reply:
nope. Consider the first iteration. we need to set both; leftPt and rightPt, to tmpPt